client/app/components/CreateEventModal.jsx

```diff
@@ -0,0 +1,181 @@
+                type="text"
+                value={this.state.description}
+                onChange={this._handleTextFieldChange}
+                name="description" />
+            </FormGroup>
+            <FormGroup>
+              <ControlLabel>Start time:</ControlLabel>
+              <DateTimeField
+                dateTime={this.state.startTime}
+                format={DATE_FORMAT}
+                inputFormat={DATE_FORMAT}
+                onChange={this._handleStartTimeChange} />
+            </FormGroup>
+            <FormGroup>
+              <ControlLabel>End time:</ControlLabel>
+              <DateTimeField
+                dateTime={this.state.endTime}
+                format={DATE_FORMAT}
+                inputFormat={DATE_FORMAT}
+                onChange={this._handleEndTimeChange} />
+            </FormGroup>
+            <FormGroup>
+              <ControlLabel>Location:</ControlLabel>
+              <FormControl
+                componentClass="select"
+                onChange={this._handleLocationChange}
+                name="location">
+                {conferenceRoomsOptions}
+              </FormControl>
+            </FormGroup>
+          </form>
+        </Modal.Body>
+
+        <Modal.Footer>
+          <Button onClick={this.props.closeModal}>Close</Button>
+          <Button
+            bsStyle="primary"
+            onClick={this._saveChanges}>
+            Save changes
+          </Button>
+        </Modal.Footer>
+      </Modal>
+    );
+  }
+
+
+  _handleTextFieldChange(e) {
+    const value = e.target.value;
+    const name = e.target.name;
+
+    this.setState({ [name]: value });
+  }
+
+  _handleLocationChange(e) {
+    const conferenceRoomId = e.target.value;
+
+    // only temporary, waiting for event API improvement
+    const conferenceRoomName = this.props.conferenceRooms.find(room =>
+      room.id === parseInt(conferenceRoomId, 10)
+    ).title;
+
+    this.setState({
+      location: conferenceRoomName,
+      conferenceRoomId
+    });
+  }
+
+  _handleStartTimeChange(e) {
+    if (e !== 'Invalid date') {
+      this.setState({ startTime: e });
+    }
+  }
+
+  _handleEndTimeChange(e) {
+    if (e !== 'Invalid date') {
+      this.setState({ endTime: e });
+    }
+  }
+
+  _showError() {
+    this.setState({ showErrorMessage: true });
+  }
+
+  _saveChanges() {
+    const eventParams = {
+      summary:            this.state.summary ? this.state.summary : '',
+      description:        this.state.description ? this.state.description : '',
+      start_time:         this.state.startTime,
+      end_time:           this.state.endTime,
+      conference_room_id: this.state.conferenceRoomId,
+      location:           this.state.location
+    };
+
+    const token = document.querySelector('meta[name="csrf-token"]').content;
+
+    axios({
+      method:  'post',
+      url:     '/events',
+      data:    eventParams,
+      headers: {
+        'X-CSRF-Token': token
+      }
+    })
```

Comment:
If you look into Refresh button #68 pull request I added an EventSource file.
I think it's a good convention to just provide a simple API to components for requests.
Also it would be cleaner 'cuz then there would be a single place where you're handling the token.